mysql-schema/035-webauthn-registration.sql

```diff
@@ -0,0 +1,21 @@
+SET NAMES utf8mb4;
+START TRANSACTION;
+
+INSERT INTO changelog VALUES (35, UNIX_TIMESTAMP());
+
+CREATE TABLE user_webauthn (
+  id INT UNSIGNED NOT NULL PRIMARY KEY AUTO_INCREMENT,
+  user_id INT UNSIGNED NOT NULL,
+  credential_id VARCHAR(2000) NOT NULL,
+  public_key VARCHAR(2000) NOT NULL,
+  counter INT UNSIGNED NOT NULL,
+  created INT UNSIGNED NOT NULL
+);
+
+INSERT INTO server_settings (setting, value) VALUES
+('registration.mfa.enabled', 'false'),
+('webauthn.relyingPartyId', '"localhost"'),
```

Comment:
Would it be possible to infer this? It will be a bit of a pain to have people update it.

Reply:
https://www.w3.org/TR/webauthn/#relying-party-identifier
> Note: An RP ID is based on a host's domain name. It does not itself include a scheme or port, as an origin does. The RP ID of a public key credential determines its scope. I.e., it determines the set of origins on which the public key credential may be exercised, as follows:
> - The RP ID must be equal to the origin's effective domain, or a registrable domain suffix of the origin's effective domain.
> - The origin's scheme must be https.
> - The origin's port is unrestricted.
> 
> For example, given a Relying Party whose origin is https://login.example.com:1337, then the following RP IDs are valid: login.example.com (default) and example.com, but not m.login.example.com and not com.
> 
> This is done in order to match the behavior of pervasively deployed ambient credentials (e.g., cookies, [RFC6265]). Please note that this is a greater relaxation of "same-origin" restrictions than what document.domain's setter provides.

No, this unfortunately cannot be inferred since the server won't know what domain it belongs to. Maybe this would work better with a different default value? 

Using the examples in this quote, `login.example.com` would be the Relying Party ID and `https://login.example.com:1337` would be the expected origin.

The `How does user registration work ?` section on https://www.okta.com/blog/2019/05/the-ultimate-guide-to-webauthn-registration-and-auth-flows/ explains this more simply.

Reply:
It's definitely going to be a bit of a pain to default to localhost, as this will not work out of the box for most people. I think in that case, it might be better to default to *no value* and report an error in the log that MFA will only be enabled once this value is set.

Reply:
Yeah.... Definitely is a shame. It's why I defaulted `registration.mfa.enabled` to `false`. I'll set it to no value and log some errors as you suggested.

Reply:
Yea similar to the SMTP setting, I think we also default to *not* having a database value, and then echo something on the console along the lines off: "We're not sending email until you put something in the settings table for this"